holoviews/plotting/util.py

```diff
@@ -72,6 +73,67 @@ def collate(obj):
         raise Exception(undisplayable_info(obj))
 
 
+def isoverlay_fn(obj):
+    """
+    Determines whether object is a DynamicMap returning (Nd)Overlay types.
+    """
+    return isinstance(obj, DynamicMap) and (isinstance(obj.last, CompositeOverlay))
+
+
+def compute_overlayable_zorders(obj, path=[]):
+    """
+    Traverses the Callables on DynamicMap to determine which objects
+    in the graph are associated with specific (Nd)Overlay layers.
+    Returns a mapping between the zorders of each layer and lists of
+    objects associated with each.
+
+    Used to determine which subplots should be linked with Stream
+    callbacks.
+    """
+    path = path+[obj]
+    zorder_map = defaultdict(list)
+
+    # Process non-dynamic layers
+    if not isinstance(obj, DynamicMap):
+        if isinstance(obj, CompositeOverlay):
+            for i, o in enumerate(obj):
+                zorder_map[i] = [o, obj]
+        else:
+            if obj not in zorder_map[0]:
+                zorder_map[0].append(obj)
+        return zorder_map
+
+    isoverlay = isinstance(obj.last, CompositeOverlay)
+    isdynoverlay = obj.callback._is_overlay
+    found = any(isinstance(p, DynamicMap) and p.callback._overlay for p in path)
+    if obj not in zorder_map[0] and not isoverlay:
+        zorder_map[0].append(obj)
```

Comment:
I think this would execute if ``.last`` is None. Maybe this is ok because you are assuming ``initial_key`` is called already?